// boardgamegeekclient/dist/cjs/client/concrete/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BggHotClient = exports.BggSearchClient = exports.BggCollectionClient = exports.BggPlayClient = exports.BggGuildClient = exports.BggUserClient = exports.BggThreadClient = exports.BggForumClient = exports.BggForumlistClient = exports.BggFamilyClient = exports.BggThingClient = void 0;
var BggThingClient_1 = require("./BggThingClient");
Object.defineProperty(exports, "BggThingClient", { enumerable: true, get: function () { return BggThingClient_1.BggThingClient; } });
var BggFamilyClient_1 = require("./BggFamilyClient");
Object.defineProperty(exports, "BggFamilyClient", { enumerable: true, get: function () { return BggFamilyClient_1.BggFamilyClient; } });
var BggForumlistClient_1 = require("./BggForumlistClient");
Object.defineProperty(exports, "BggForumlistClient", { enumerable: true, get: function () { return BggForumlistClient_1.BggForumlistClient; } });
var BggForumClient_1 = require("./BggForumClient");
Object.defineProperty(exports, "BggForumClient", { enumerable: true, get: function () { return BggForumClient_1.BggForumClient; } });
var BggThreadClient_1 = require("./BggThreadClient");
Object.defineProperty(exports, "BggThreadClient", { enumerable: true, get: function () { return BggThreadClient_1.BggThreadClient; } });
var BggUserClient_1 = require("./BggUserClient");
Object.defineProperty(exports, "BggUserClient", { enumerable: true, get: function () { return BggUserClient_1.BggUserClient; } });
var BggGuildClient_1 = require("./BggGuildClient");
Object.defineProperty(exports, "BggGuildClient", { enumerable: true, get: function () { return BggGuildClient_1.BggGuildClient; } });
var BggPlaysClient_1 = require("./BggPlaysClient");
Object.defineProperty(exports, "BggPlayClient", { enumerable: true, get: function () { return BggPlaysClient_1.BggPlayClient; } });
var BggCollectionClient_1 = require("./BggCollectionClient");
Object.defineProperty(exports, "BggCollectionClient", { enumerable: true, get: function () { return BggCollectionClient_1.BggCollectionClient; } });
var BggSearchClient_1 = require("./BggSearchClient");
Object.defineProperty(exports, "BggSearchClient", { enumerable: true, get: function () { return BggSearchClient_1.BggSearchClient; } });
var BggHotClient_1 = require("./BggHotClient");
Object.defineProperty(exports, "BggHotClient", { enumerable: true, get: function () { return BggHotClient_1.BggHotClient; } });
