// boardgamegeekclient/dist/cjs/dto/concrete/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BggHotDto = exports.BggSearchDto = exports.BggCollectionDto = exports.BggPlayDto = exports.BggGuildDto = exports.BggUserDto = exports.BggThreadDto = exports.BggForumDto = exports.BggForumlistDto = exports.BggFamilyDto = exports.BggThingDto = void 0;
var BggThingDto_1 = require("./BggThingDto");
Object.defineProperty(exports, "BggThingDto", { enumerable: true, get: function () { return BggThingDto_1.BggThingDto; } });
var BggFamilyDto_1 = require("./BggFamilyDto");
Object.defineProperty(exports, "BggFamilyDto", { enumerable: true, get: function () { return BggFamilyDto_1.BggFamilyDto; } });
var BggForumlistDto_1 = require("./BggForumlistDto");
Object.defineProperty(exports, "BggForumlistDto", { enumerable: true, get: function () { return BggForumlistDto_1.BggForumlistDto; } });
var BggForumDto_1 = require("./BggForumDto");
Object.defineProperty(exports, "BggForumDto", { enumerable: true, get: function () { return BggForumDto_1.BggForumDto; } });
var BggThreadDto_1 = require("./BggThreadDto");
Object.defineProperty(exports, "BggThreadDto", { enumerable: true, get: function () { return BggThreadDto_1.BggThreadDto; } });
var BggUserDto_1 = require("./BggUserDto");
Object.defineProperty(exports, "BggUserDto", { enumerable: true, get: function () { return BggUserDto_1.BggUserDto; } });
var BggGuildDto_1 = require("./BggGuildDto");
Object.defineProperty(exports, "BggGuildDto", { enumerable: true, get: function () { return BggGuildDto_1.BggGuildDto; } });
var BggPlayDto_1 = require("./BggPlayDto");
Object.defineProperty(exports, "BggPlayDto", { enumerable: true, get: function () { return BggPlayDto_1.BggPlayDto; } });
var BggCollectionDto_1 = require("./BggCollectionDto");
Object.defineProperty(exports, "BggCollectionDto", { enumerable: true, get: function () { return BggCollectionDto_1.BggCollectionDto; } });
var BggSearchDto_1 = require("./BggSearchDto");
Object.defineProperty(exports, "BggSearchDto", { enumerable: true, get: function () { return BggSearchDto_1.BggSearchDto; } });
var BggHotDto_1 = require("./BggHotDto");
Object.defineProperty(exports, "BggHotDto", { enumerable: true, get: function () { return BggHotDto_1.BggHotDto; } });
